basculechecks/capabilitiesmap.go

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright 2020 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package basculechecks
+
+import (
+	"errors"
+
+	"github.com/goph/emperror"
+	"github.com/xmidt-org/bascule"
+)
+
+var (
+	ErrNilDefaultChecker = errors.New("default checker cannot be nil")
+	ErrEmptyEndpoint     = errors.New("endpoint provided is empty")
+)
+
+// CapabilitiesMap runs a capability check based on the value of the parsedURL,
+// which is the key to the CapabilitiesMap's map.  The parsedURL is expected to
+// be some regex values, allowing for bucketing of urls that contain some kind
+// of ID or otherwise variable portion of a URL.
+type CapabilitiesMap struct {
+	checkers       map[string]CapabilityChecker
+	defaultChecker CapabilityChecker
+}
+
+// NewCapabilitiesMap validates the default Checker and sets up the map to be
+// non-nil in order to create a good CapabilitiesMap struct.
+func NewCapabilitiesMap(checkers map[string]CapabilityChecker, defaultChecker CapabilityChecker) (*CapabilitiesMap, error) {
+	if defaultChecker == nil {
```

Comment:
Don't forget that a `nil` map or slice is valid for reading, just not for writing.  You can allow `nil` to let your client not bother creating things s/he doesn't need.

https://play.golang.org/p/fZ4Oo_ofkaE